harpy/bin/inline_to_haplotag.py

```
@@ -55,24 +55,44 @@ def validate_barcode(barcode):
+    # search for a valid barcode at all possible barcode lengths
+    if fw_entry:
+        for length in bc_lengths:
+            bc_len = length
+            bc_inline = fw_entry[1][:length]
+            bc_hap = barcode_dict.get(bc_inline, "A00C00B00D00")
+            # end the loop if the barcode was found in the dict, which would return None or a valid ACBD barcode
+            if bc_hap != "A00C00B00D00":
+                break
+        # the default barcode entry is None, meaning it hasnt been assigned a haplotag equivalent yet
+        if not bc_hap:
+            bchap = "".join(next(haplotag_bc))
```

Ruff

Local variable `bchap` is assigned to but never used

Remove assignment to unused variable `bchap`

(F841)